controllers/control_plane_init_locker.go

```diff
@@ -77,55 +81,55 @@ func (l *controlPlaneInitLocker) Acquire(cluster *clusterv2.Cluster) bool {
 				},
 			},
 		},
+		Data: map[string]string{
+			lockInformationKey: string(b),
+		},
 	}
 
 	log.Info("Attempting to create control plane configmap lock")
-	if err := l.client.Create(l.ctx, controlPlaneConfigMap); err != nil {
-		if apierrors.IsAlreadyExists(err) {
-			// Someone else beat us to it
-			log.Info("Control plane configmap lock already exists")
-		} else {
-			log.Error(err, "Error creating control plane configmap lock")
-		}
-
-		// Unable to acquire
+	err = c.client.Create(ctx, controlPlaneConfigMap)
+	switch {
+	case apierrors.IsAlreadyExists(err):
+		log.Info("Control plane configmap lock already exists")
+		return false
+	case err != nil:
+		log.Error(err, "Error creating control plane configmap lock")
 		return false
+	default:
+		return true
 	}
-
-	// Successfully acquired
-	return true
 }
 
-func (l *controlPlaneInitLocker) Release(cluster *clusterv2.Cluster) bool {
+// Release releases the lock
+func (c *ControlPlaneInitLocker) Release(ctx context.Context, cluster *clusterv1.Cluster) bool {
 	configMapName := fmt.Sprintf("%s-controlplane", cluster.UID)
```

Comment:
Should this match [L56](https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/pull/166/files#diff-35f2df2255633bb300bab83f5d0953d6R56)?

Reply:
yes it should!